### ISSUE TITLE
Convert yolov4.weight to TensorRT

### ISSUE THREAD

**Issue by theojaasp** [jkjung-avt/jkjung-avt.github.io#1]
Hello, I have a short question. I try to convert the yolov4 weight file but whatever I do the memory is not enough. So is it possible to convert the file on a other machine, and if yes what did the machine require to convert? 

Regards

**Comment by jkjung-avt** [comment]
Yes.  It is OK to convert DarkNet model to ONNX ("yolo_to_onnx.py") on another machine.

But you must compile the plugin and do "onnx_to_tensorrt.py" on the target platform.

**Comment by theojaasp** [comment]
Okay, thanks for the answer. I did it successfully.
Is there any drawback in the detection of the object? Because we have less inference(more fps) ?
Did you have some comments on the detection precision? And did you have any instruction how to trains models? Or can I normally go on with train on darknet yolo and convert to TensorRT?

**Comment by jkjung-avt** [comment]
> Is there any drawback in the detection of the object?

No.  As long as you use the same version (1.4.1) of python3 "onnx" module, the converted ONNX file should be exactly the same.

> Did you have some comments on the detection precision? And did you have any instruction how to trains models? Or can I normally go on with train on darknet yolo and convert to TensorRT?

You could refer to my [yolov4_crowdhuman](https://github.com/jkjung-avt/yolov4_crowdhuman) tutorial, in which I demonstrated how to train a DarkNet yolov4 model and how to convert it to a TensorRT engine.

**Comment by theojaasp** [comment]
Great thank you. I appreciate your answer.

Have you any information about the fps difference betweeen reading images, videos or live from usb camera?
Because I have noticed that when I use the dog.jpg picture It stays in area about 52.7 fps, but when using an example video the fps starts at 35.6 fps and decrease to around 12.7 fps. When I am reading through my USB webcam it starts at 27.8 fps and decrease to around 11.3 fps. But when I am looking at the video I can not say that the fps matches to the output I can see on my camera. Can it be that there is something wrong in the fps calculation or is that possible that I really decreasing performance during the detection?

**Comment by jkjung-avt** [comment]
> Have you any information about the fps difference between reading images, videos or live from usb camera?

My [tensorrt_demos](https://github.com/jkjung-avt/tensorrt_demos) code utilizes "cv2" and gstreamer pipelines to capture video frames.  So FPS would not only depend on the video source your are using, as well as your cv2 (libopencv) and gstreamer libraries.

To get the best performance for your platform, you might consider rewriting the video acquisition part of the code.

**Comment by theojaasp** [comment]
> > Have you any information about the fps difference between reading images, videos or live from usb camera?
> 
> My [tensorrt_demos](https://github.com/jkjung-avt/tensorrt_demos) code utilizes "cv2" and gstreamer pipelines to capture video frames. So FPS would not only depend on the video source your are using, as well as your cv2 (libopencv) and gstreamer libraries.
> 
> To get the best performance for your platform, you might consider rewriting the video acquisition part of the code.

**Comment by theojaasp** [comment]
> My [tensorrt_demos](https://github.com/jkjung-avt/tensorrt_demos) code utilizes "cv2" and gstreamer pipelines to capture video frames. So FPS would not only depend on the video source your are using, as well as your cv2 (libopencv) and gstreamer libraries.
>
I am using the Jetson Nano Dev kit for this project. So I have opencv, TensorRT, CUDA and cuDNN pre -installed.

> To get the best performance for your platform, you might consider rewriting the video acquisition part of the code.
>
I will look what I can do. I am not much confident in writing code.
